src/modules/app/containers/Switcher.js

```diff
@@ -0,0 +1,111 @@
+import DisclaimerPage from '../../../routes/disclaimer/containers/DisclaimerPage'
+import SearchPage from '../../../routes/search/containers/SearchPage'
+import PdfFetcherPage from '../../../routes/pdf-fetcher/containers/PdfFetcherPage'
+import { LANDING_ROUTE } from '../routes/landing'
+import { MAIN_DISCLAIMER_ROUTE } from '../routes/mainDisclaimer'
+import { CATEGORIES_ROUTE } from '../routes/categories'
+import { EVENTS_ROUTE } from '../routes/events'
+import { EXTRAS_ROUTE } from '../routes/extras'
+import { DISCLAIMER_ROUTE } from '../routes/disclaimer'
+import { SEARCH_ROUTE } from '../routes/search'
+import { PDF_FETCHER_ROUTE } from '../routes/pdfFetcher'
+
+import Payload from '../../endpoint/Payload'
+import Failure from '../../common/components/Failure'
+
+type Props = {
+  viewportSmall: boolean,
+  currentRoute: string,
+  citiesPayload: Payload,
+  categoriesPayload: Payload,
+  eventsPayload: Payload,
+  extrasPayload: Payload,
+  disclaimerPayload: Payload
+}
+
+/**
+ * Renders different Pages depending on the current route. If the needed data is not available, a LoadingSpinner is rendered
+ */
+class Switcher extends React.Component<Props> {
+  static getDisplayComponent (page: React.Node, payload: Payload): React.Node {
+    const LoadingSpinner = () => <Spinner name='line-scale-party' />
+    if (payload.isFetching) {
+      return <LoadingSpinner />
+    } else if (payload.data) {
+      return page
+    } else if (payload.error) {
+      return <Failure error={payload.error} />
+    } else {
+      return null
+    }
+  }
+
+  getPage (): React.Node {
+    const {currentRoute, citiesPayload, eventsPayload, categoriesPayload, extrasPayload, disclaimerPayload} = this.props
+
+    switch (currentRoute) {
+      case LANDING_ROUTE:
+        return Switcher.getDisplayComponent(<LandingPage />, citiesPayload)
+      case MAIN_DISCLAIMER_ROUTE:
+        return <MainDisclaimerPage />
+      case CATEGORIES_ROUTE:
+        return Switcher.getDisplayComponent(<CategoriesPage />, categoriesPayload)
+      case EVENTS_ROUTE:
+        return Switcher.getDisplayComponent(<EventsPage />, eventsPayload)
+      case EXTRAS_ROUTE:
+        return Switcher.getDisplayComponent(<ExtrasPage />, extrasPayload)
+      case DISCLAIMER_ROUTE:
+        return Switcher.getDisplayComponent(<DisclaimerPage />, disclaimerPayload)
+      case SEARCH_ROUTE:
+        return Switcher.getDisplayComponent(<SearchPage />, categoriesPayload)
+      case PDF_FETCHER_ROUTE:
+        return Switcher.getDisplayComponent(<PdfFetcherPage />, categoriesPayload)
+    }
+    return null
+  }
+
+  render () {
+    const {viewportSmall, currentRoute} = this.props
+
+    if (currentRoute === LANDING_ROUTE) {
+      return <Layout footer={<GeneralFooter />}>
+        {this.getPage()}
+      </Layout>
+    } else if ([CATEGORIES_ROUTE, EVENTS_ROUTE, DISCLAIMER_ROUTE, EXTRAS_ROUTE, SEARCH_ROUTE].includes(currentRoute)) {
```

Comment:
I think it would be cool if this would be some kind of configuration like:
```
{
    render: () => <EventsPage />
    layout: LOCATION_LAYOUT
}
```

Reply:
What do you mean with render: () => <EventsPage />?

Reply:
I don't understand it :/

Reply:
The render function would return the node which needs to be renders for this route.

Reply:
@maxammann you could add a task